### PR TITLE
CLOUDP-156204: remove modules and cleanup folder structure

### PR DESCRIPTION
**Pull request by wtrocki**
## Description

Introduce folder and user experience changes according to the spec document. 
Add manually written readme to demo end-user experience.

## Notable changes

- Added readme that will be supported by go doc
- tools and open API folder moved out of the v2 source code (based on previous PRs and feedback)
- v1 folder instead of v2 folder (GA version of the SDK, assumes that current code is v0)

> NOTE: I did not move any static files like generated code as this will generate +600 files changed PR. 
I will do that in a separate PR that will not require many reviews. 

> NOTE: Reasons behind the changes have been explained in the spec.

### CODE REVIEW

File: api/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+```go
+	mongodbatlas import "go.mongodb.org/atlas/mongodbatlas/api/v1"
+
+   	apiKey := os.Getenv("MDB_API_KEY")
+	apiSecret := os.Getenv("MDB_API_SECRET")
+
+	// SDK client with authentication
+	sdk := mongodbatlas.NewSDKClientWithCredentials(apiKey, apiSecret)
+    projects, response, err := sdk.ProjectsApi.ListProjects(ctx).Execute()
```
Comment (gssbzn) [inline api/README.md:30]: seems like a small indentation issue here, ie spaces vs tabs

File: go.mod [hunk 2]
```diff
@@ -3,7 +3,16 @@ module go.mongodb.org/atlas
...
 	github.com/google/go-querystring v1.1.0
 	github.com/openlyinc/pointy v1.2.0
+	
+	// v1 dependencies
+	github.com/mongodb-forks/digest v1.0.4
+	golang.org/x/oauth2 v0.0.0-20210323180902-22b0adad7558
+	github.com/golang/protobuf v1.4.2 // indirect
+	golang.org/x/net v0.0.0-20200822124328-c89045814202 // indirect
+	google.golang.org/appengine v1.6.6 // indirect
+	google.golang.org/protobuf v1.25.0 // indirect
```
Comment (gssbzn) [inline go.mod:17]: since go1.17 indirect dependencies go in a separate group

File: go.sum [hunk 3]
```diff
@@ -8,5 +9,11 @@ github.com/google/go-querystring v1.1.0/go.mod h1:Kcdr2DB4koayq7X8pmAG4sNG59So17
...
 github.com/pmezard/go-difflib v1.0.0 h1:4DBwDE0NGyQoBHbLQYPwSUPoCMWR5BEzIk/f1lZbAQM=
-github.com/stretchr/testify v1.2.2 h1:bSDNvY7ZPG5RlJ8otE/7V6gMiyenm9RtJ7IUVIAoJ1w=
+github.com/pmezard/go-difflib v1.0.0/go.mod h1:iKH77koFhYxTK1pcRnkKkqfTogsbg7gZNVY4sRDYZ/4=
+github.com/stretchr/objx v0.1.0/go.mod h1:HFkY916IF+rwdDfMAkV7OtwuqBVzrE8GR6GFx+wExME=
+github.com/stretchr/testify v1.4.0 h1:2E4SXV/wtOkTonXsotYi4li6zVWxYlZuYNCXe9XRJyk=
+github.com/stretchr/testify v1.4.0/go.mod h1:j7eGeouHqKxXV5pUuKE4zz7dFj8WfuZ+81PSLYec5m4=
 golang.org/x/xerrors v0.0.0-20191204190536-9bdfabe68543/go.mod h1:I/5z698sn9Ka8TeJc9MKroUUfqBBauWjQqLJ2OPfmY0=
+gopkg.in/check.v1 v0.0.0-20161208181325-20d25e280405/go.mod h1:Co6ibVJAznAaIkqp8huTwlJQCZ016jof/cbN4VW5Yz0=
+gopkg.in/yaml.v2 v2.2.2 h1:ZCJp+EgiOT7lHqUV2J862kp8Qj64Jo6az82+3Td9dZw=
+gopkg.in/yaml.v2 v2.2.2/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
```
Comment (gssbzn) [inline go.sum:19]: `gopkg.in/yaml.v2` is deprecated and unsafe, I supoose this is the dep of a dep but can you verify what's including it to chekc if we can exclude it?

File: api/README.md (hunk 1 above)
Comment (wtrocki) [reply api/README.md:30]: Good catch. 
I will consider setting up prettier in the repository to catch up issues like this.

File: api/README.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+access different parts of the Atlas API. For example:
+
+```go
+	mongodbatlas import "go.mongodb.org/atlas/mongodbatlas/api/v1"
+
+   	apiKey := os.Getenv("MDB_API_KEY")
+	apiSecret := os.Getenv("MDB_API_SECRET")
+
+	// SDK client with authentication
+	sdk := mongodbatlas.NewSDKClientWithCredentials(apiKey, apiSecret)
```
Comment (gssbzn) [inline api/README.md:29]: as a follow up to this PR but these constructs with discrete options don't scale well, if we want to offer these wrappers let's follow a functional approach like the current client does

https://github.com/mongodb/go-client-mongodb-atlas/blob/44ada01aee87c1ba1a6897b492117863281487d3/mongodbatlas/mongodbatlas.go#L312-L333

**Review by gssbzn — COMMENTED** [review]
some small nits, one spacing isssue on the readme and to verify some questions I left for the go.mod/go.sum

File: api/README.md (hunk 4 above)
Comment (wtrocki) [reply api/README.md:29]: Great idea!
It will also make things easier to migrate.

File: go.sum (hunk 3 above)
Comment (wtrocki) [reply go.sum:19]: This was introduced in a pointy package that is not used outside testing.

File: api/README.md (hunk 4 above)
Comment (wtrocki) [reply api/README.md:29]: Added separate jira as adding it will trigger number of extra files changed.

Comment (gssbzn) [reply api/README.md:32]: agreed

**Review by gssbzn — APPROVED** [review]

**Pull request merged**